doc/smart-switch/high-availability/smart-switch-ha-hld.md

```diff
@@ -0,0 +1,2176 @@
+
+<p align="center"><img src="./images/ha-active-standby-setup.svg"></p>
+
+### 4.2. Data path setup
+
+Besides providing HA for flows, we also need to provide solutions to ensure data path works with HA considered:
+
+1. Ensures having enough redundant links for the traffic to reach its target HA set.
+2. Ensures having enough redundant links for the 2 DPUs in the HA set to talk to each other.
+
+And this can be addressed by the ECMP setup in our network, as the "[Network topology for Flow HA](#network-topology-for-flow-ha)" section shows above.
+
+
+#### 4.2.1. Upstream-to-Switch communication channel
+
+In HA setup, our upstream service will talk to our northbound interface for:
+
+- Programming SDN dash policies.
+- Initiate the planned operation, or unplanned operation which can be destructive.
+- Get HA states and receive the HA state reports.
+- Setting up the traffic forwarding tunnels.
+
+#### 4.2.2. NPU-to-DPU traffic forward tunnel
+
+This channel is used for tunneling the packet from all T1s that has the traffic forwarding rules to the active side in the HA pair, no matter if it is locally or remotely.
+
+There is a limitation on the traffic forwarding rules, it can only forward the traffic to either local DPU or remote DPU, but never both as an ECMP group. This is caused by the limitation of routes. Therefore, the tunnel will always only have 1 next hop at maximum.
+
+It works as below in high level:
+
+1. The tunneled packets ***MUST*** have an VxLan encap as the outer packet, with a specified NPU tunnel VNI number as identifier.
+2. The outer packet ***MUST*** calculate the source port from the 5 tuples of the inner packet and fold it into a port range specified by SONiC.
+3. The outer packet ***MUST*** have the same DSCP fields as the inner packet.
+
+The detailed data path is shown below.
+
+##### 4.2.2.1. Tunneling packet to local DPU
+
+<p align="center"><img alt="Tunneling packet to local DPU" src="./images/npu-to-dpu-tunnel-local.svg"></p>
+
+##### 4.2.2.2. Tunneling packet to remote DPU
+
+<p align="center"><img alt="Tunneling packet to remote DPU" src="./images/npu-to-dpu-tunnel-remote.svg"></p>
+ 
+##### 4.2.2.3. Tunneling packet between DPUs
+
+This tunnel needs to be also implemented by DPU, because in planned or unplanned events, traffic might land on the in non-active side DPU and needs to be tunneled. This is using the same NPU-to-DPU traffic tunnel.
+ 
+<p align="center"><img alt="Tunneling packet between DPUs" src="./images/npu-to-dpu-tunnel-from-dpu.svg"></p>
+
+##### 4.2.2.4. Put all tunnels together
+
+Now let’s put all tunnels together. This can happen when the packets land on other switches that help traffic forwarding in the middle of HA state transition.
+ 
+<p align="center"><img alt="Tunneling packet between DPUs" src="./images/npu-to-dpu-tunnel-2-stages.svg"></p>
+
+#### 4.2.3. DPU-to-DPU communication channel
+
+There are 3 types of DPU-to-DPU communication channel: 
```

Comment:
it's my understanding that some of these channels will have standardized formats/protocols and some may have vendor-specific protocols (e.g. inter-DPU flow sync). Can you add commentary about this?

Reply:
this is a very good question. for the 3 channels, the detailed packet format of HA control channel and DPU data plane channel are shown in the graph, which contains the information. and I have updated the HA data channel.